packages/x-tree-view/src/useTreeItem2/useTreeItem2.types.ts

```diff
@@ -81,11 +82,23 @@ export type UseTreeItemIconContainerSlotProps<ExternalProps = {}> = ExternalProp
 
 export interface UseTreeItem2LabelSlotOwnProps {
   children: React.ReactNode;
+  onDoubleClick: MuiCancellableEventHandler<React.MouseEvent>;
+  editable?: boolean;
```

Comment:
```suggestion
  /**
   * Only defined when the `isItemEditable` experimental feature is enabled.
   */
  editable?: boolean;
```

+ modify the place where you create this prop to actually follow this rule

This way people with custom label slots won't have an invalid props passed to their DOM (this is a super small BC).
See `indentationAtItemLevel` on `UseTreeItem2ContentSlotPropsFromUseTreeItem` for reference.